subsys/bluetooth/controller/hci/hci.c

```diff
@@ -2816,6 +2835,8 @@ void hci_acl_encode(struct radio_pdu_node_rx *node_rx, struct net_buf *buf)
 			LL_ASSERT((hci_hbuf_sent - hci_hbuf_acked) <
 				  hci_hbuf_total);
 			hci_hbuf_sent++;
+			LL_ASSERT(handle < ARRAY_SIZE(hci_hbuf_pend));
+			hci_hbuf_pend[handle]++;
```

Comment:
Please have a TODO or NOTE, to mention that connection handles could be random from controller (not in the current ll_sw controller implementation though), and a lookup table implementation is needed.